### ISSUE TITLE
screensaver stay white after return from sleep (using background mode)

### ISSUE THREAD

**Issue by Waitsnake** [Waitsnake/AnimatedGif#15]
There is an issue that glview is not working (stay white) after return from sleep when using background mode.

**Comment by Waitsnake** [comment]
A working solution comes with commit 47.

**Comment by Waitsnake** [comment]
This is no issue of glview as I thought before.

I was analyzing the problem a bit deeper and it starts with bugfix of issue #14 . The call of ScreenSaverEngine with parameter "-idleCheck" that we used there to have two instances of ScreenSaverEngine(one background and one foreground) leeds to the situation that ScreenSaverEngine not receives any longer the notifications that are needed to handle sleep situation correct. This causes ScreenSaverEngine to an "unexpected screen saver end" that can be found in the console log and leeds to white screen of this issue. So killing the already crashed ScreenSaverEngine seems at the moment the best workaround if we still want a solution for issue #14 . Even if this is not a clean solution and I don't like it very much. You will still find those "unexpected screen saver end" in the console log.

The old release 1.1 that have no fix for issue #14 don't have the issue #15 either. But I found out that there is a similar problem. When going to sleep a couple of times (most times it works) there is a small possibility that screensaver hangs too and screen become black (and not white). In this situation there is no "unexpected screen saver end" in the console log, but screensaver also not works any longer. I guess there is another bug in the ScreenSaverEngine that sleeping is not covered correct all times. The fix for #15 should cover the black screen too.
